discovery/v1.js

```diff
@@ -255,14 +255,23 @@ DiscoveryV1.prototype.addDocument = function(params, callback) {
   params = params || {};
 
   var query_params = pick(params, ['configuration_id'])
+  var formDataParams = pick(params, ['file', 'metadata']);
+
+  // if we get a buffer or object, we need to include stuff about filename for the service
+  if (formDataParams.file) {
+    if (typeof formDataParams.file.filename !== 'string') {
```

Comment:
fixed.

The filename can be inferred from file streams, but not if they're a buffer.